app/src/main/java/org/simple/clinic/teleconsultlog/drugduration/DrugDurationSheet.kt

```diff
@@ -0,0 +1,156 @@
+        drugDuration: String
+    ): Intent {
+      return Intent(context, DrugDurationSheet::class.java).apply {
+        putExtra(EXTRA_DRUG, drug)
+        putExtra(EXTRA_DURATION, drugDuration)
+      }
+    }
+  }
+
+  private lateinit var component: DrugDurationComponent
+
+  private val prescribedDrug by unsafeLazy {
+    intent.getParcelableExtra<PrescribedDrug>(EXTRA_DRUG)!!
+  }
+
+  private val events by unsafeLazy {
+    Observable
+        .merge(
+            imeClicks(),
+            durationChanges()
+        )
+        .compose(ReportAnalyticsEvents())
+  }
+
+  private val delegate by unsafeLazy {
+    val duration = intent.getStringExtra(EXTRA_DURATION)!!
+
+    MobiusDelegate.forActivity(
+        events = events.ofType(),
+        defaultModel = DrugDurationModel.create(duration),
+        init = DrugDurationInit(),
+        update = DrugDurationUpdate(),
+        effectHandler = effectHandlerFactory.create(this).build()
+    )
+  }
+
+  override fun onStart() {
+    super.onStart()
+    delegate.start()
+  }
+
+  override fun onStop() {
+    delegate.stop()
+    super.onStop()
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.sheet_drug_duration)
+    delegate.onRestoreInstanceState(savedInstanceState)
+
+    drugDurationTitleTextView.text = getString(R.string.drug_duration_title, prescribedDrug.name, prescribedDrug.dosage)
+  }
+
+  override fun onSaveInstanceState(outState: Bundle) {
+    super.onSaveInstanceState(outState)
+    delegate.onSaveInstanceState(outState)
+  }
+
+  override fun attachBaseContext(baseContext: Context) {
+    setupDi()
+
+    val wrappedContext = baseContext
+        .wrap { LocaleOverrideContextWrapper.wrap(it, locale) }
+        .wrap { InjectorProviderContextWrapper.wrap(it, component) }
+        .wrap { ViewPumpContextWrapper.wrap(it) }
+
+    super.attachBaseContext(wrappedContext)
+  }
+
+  private fun setupDi() {
+    component = ClinicApp.appComponent
+        .drugDurationComponent()
+        .activity(this)
+        .build()
+
+    component.inject(this)
+  }
+
+  private fun imeClicks(): Observable<DrugDurationEvent> {
+    return drugDurationEditText
+        .editorActions { it == EditorInfo.IME_ACTION_DONE }
+        .map { DrugDurationSaveClicked(drugDurationEditText.text.toString()) }
+  }
+
+  private fun durationChanges(): Observable<DrugDurationEvent> {
+    return drugDurationEditText
+        .textChanges { DurationChanged(it) }
+  }
+
+  override fun showBlankDurationError() {
+    drugDurationErrorTextView.text = getString(R.string.drug_duration_empty_error)
+    drugDurationErrorTextView.visibility = View.VISIBLE
+  }
+
+  override fun hideDurationError() {
+    drugDurationErrorTextView.text = null
+    drugDurationErrorTextView.visibility = View.GONE
+  }
+
+  override fun saveDrugDuration(duration: Int) {
+    val intent = Intent()
+    intent.putExtra(EXTRA_SAVED_DURATION, duration)
```

Comment:
Won't we also need to add the `UUID` of the `PrescribedDrug` so that the caller knows which one needs to be updated?